### PR TITLE
Adding jinja2 extension

### PR DESCRIPTION
**Pull request by destos**
Adds support for using the render_bundle function inside Jinja2.

### CODE REVIEW

**Comment by owais** [comment]
@destos Wonderful. Thanks for the PR. Can we move the section in docs to the bottom of the page inside a section called extras? I would like the main docs to stay to the point for 90% of the people.

**Comment by owais** [comment]
@destos Also, can we also reduce the configuration example to the relevant part please? Something like this,

```
TEMPLATES = [
    {
        "BACKEND": "django_jinja.backend.Jinja2",
        "OPTIONS": {
            "extensions": [
                "webpack_loader.contrib.jinja2ext.WebpackExtension",
            ],
        }
    }
]
```

and then may be link to django-jinja docs for configuration specific to setting up Jinja2 with django? Also, let mention this is for django1.8 and above.

**Comment by destos** [comment]
@owais Made the changes, also fixed an issue I found in the template tag.

File: webpack_loader/templatetags/webpack_loader.py [hunk 1]
```diff
@@ -12,10 +13,12 @@ def render_bundle(bundle_name):
...
-            tags.append('<link type="text/css" href="{}" rel="stylesheet">'.format(url))
+        path = chunk.get('path', None)
+        url = staticfiles_storage.url(path)
+        name = str(chunk.get('name'))
+        if url:
+            if name.endswith('.js'):
+                tags.append('<script type="text/javascript" src="{}"></script>'.format(url))
+            elif name.endswith('.css'):
+                tags.append('<link type="text/css" href="{}" rel="stylesheet">'.format(url))
     return '\n'.join(tags)
```
Comment (owais) [inline webpack_loader/templatetags/webpack_loader.py]: @destos I tested it and everything seems to work. BTW why did we change the URL generation code?

Comment (owais) [inline webpack_loader/templatetags/webpack_loader.py]: I mean this part shouldn't calculate the URL and chunk.publicPath is preferred to STATIC_URL path. I'll change this and merge.

**Comment by owais** [comment]
@destos Thanks a lot! This is ready. One small change. I've implemented the staticfiles integration elsewhere. Thanks for the pointer. Can you please revert your changes in `webpack_loader/templatetags/webpack_loader.py`? Thanks!

**Comment by destos** [comment]
@owais reverted the change in that file. Did you fix the issue with endswith being misstyped as endwith on the css elif check?

**Comment by owais** [comment]
@destos, thanks. I didn't fix the typo yet. I'll do that and then merge. Ive also added basic test cases. I'll add test cases for jinja myself for this one though.

**Comment by owais** [comment]
@destos I've merged your PR into master. Here https://github.com/owais/django-webpack-loader/commit/834b56367071ac97f4895f3bdb37df7c03c2ae4c

I like to follow a rebase workflow instead of merge workflow and it seems github PRs don't like that and can't figure out that the branch was merged. I think this happens because commit ids change while rebasing and squashing commits. 

I'll write some docs on how to submit squashed commits so PRs can show as merged in future.

Thanks!

**Comment by owais** [comment]
@destos Since I don't use jinja ever, I think it'll be very hard for me to maintain it. Would it be possible for you to create a support package called `django-webpack-loader-jinja` or submit a patch to django-jinja? I see they already carry patches for django-pipeline and other packages.

**Comment by owais** [comment]
The reason I'm proposing this is that I'm not sure if I can provide 100% stable jinja support. It is better to let users do it themselves or install an extra package than provide broken built in support.

**Comment by destos** [comment]
@owais, I don't see the API on Jinja's side changing that much so I would be opposed to splitting it off. I feel to support alternative template engines it's up to the original package to attempt to add support. But it of course it not required if a maintainer doesn't wish to support more than the default. 

At the moment the tests you've implemented for django_jinja should fail if there are any changes on the jinja side.

**Comment by owais** [comment]
@destos You are right, test cases should be enough for now. I think we can carry jinja2 support until it gets out of hand. 

My concern was that I'm not the best person to maintain Jinja2 support as I don't use it on a daily basis and I had to drop Python3.2 from test matrix because jinja2 doesn't work there which made the whole library drop support for 3.2. 3.2 is not a huge deal but this is an indication of what might happen in future because of jinja dependency. Anyway, thanks a lot for the patch.

**Pull request closed without merging**